File: qc-application/src/main/java/com/eager/questioncloud/application/api/payment/ChargePointPaymentHistoryService.java
package com.eager.questioncloud.application.api.payment;

import com.eager.questioncloud.core.common.PagingInformation;
import com.eager.questioncloud.core.domain.point.infrastructure.ChargePointPaymentRepository;
import com.eager.questioncloud.core.domain.point.model.ChargePointPayment;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class ChargePointPaymentHistoryService {
    private final ChargePointPaymentRepository chargePointPaymentRepository;

    public List<ChargePointPayment> getChargePointPayments(Long userId, PagingInformation pagingInformation) {
        return chargePointPaymentRepository.getChargePointPayments(userId, pagingInformation);
    }

    public int countChargePointPayment(Long userId) {
        return chargePointPaymentRepository.countByUserId(userId);
    }
}

File: qc-application/src/main/java/com/eager/questioncloud/application/business/payment/point/service/ChargePointPaymentHistoryService.kt
package com.eager.questioncloud.application.business.payment.point.service

import com.eager.questioncloud.core.common.PagingInformation
import com.eager.questioncloud.core.domain.point.infrastructure.repository.ChargePointPaymentRepository
import com.eager.questioncloud.core.domain.point.model.ChargePointPayment
import org.springframework.stereotype.Service

@Service
class ChargePointPaymentHistoryService(
    private val chargePointPaymentRepository: ChargePointPaymentRepository
) {
    fun getChargePointPayments(userId: Long, pagingInformation: PagingInformation): List<ChargePointPayment> {
        return chargePointPaymentRepository.getChargePointPayments(userId, pagingInformation)
    }

    fun countChargePointPayment(userId: Long): Int {
        return chargePointPaymentRepository.countByUserId(userId)
    }
}
